libsolidity/analysis/TypeChecker.cpp

```diff
@@ -780,8 +780,9 @@ bool TypeChecker::visit(Assignment const& _assignment)
 bool TypeChecker::visit(TupleExpression const& _tuple)
 {
 	vector<ASTPointer<Expression>> const& components = _tuple.components();
-	solAssert(!_tuple.isInlineArray(), "Tuple type not properly declared");
 	TypePointers types;
+	TypePointer internalArrayType;
```

Comment:
Please declare this closer to its use. Also the `isArray` variable is not really needed.